src/modules/test/recipes/PmcTests.json

```diff
@@ -0,0 +1,234 @@
+[
+  {
+    "ComponentName": "PackageManagerConfiguration",
+    "ObjectName": "state",
+    "Desired": false,
+    "ExpectedResult": 0
+  },
+  {
+    "ComponentName": "PackageManagerConfiguration",
+    "ObjectName": "invalidObjectName",
+    "Desired": false,
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "invalidComponentName",
+    "ObjectName": "state",
+    "Desired": false,
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "PackageManagerConfiguration",
+    "ObjectName": "",
+    "Desired": false,
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "",
+    "ObjectName": "state",
+    "Desired": false,
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "",
+    "ObjectName": "",
+    "Desired": false,
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "PackageManagerConfiguration",
+    "ObjectName": "desiredState",
+    "Desired": true,
+    "Payload": "{}",
+    "ExpectedResult": 22
+  },
+  {
+    "ComponentName": "CommandRunner",
+    "ObjectName": "commandArguments",
+    "Desired": true,
+    "Payload": "{\"commandId\":\"addMicrosoftPublicKey\",\"arguments\":\"apt-key adv --keyserver keyserver.ubuntu.com --recv-keys EB3E94ADBE1229CF\",\"timeout\":0,\"singleLineTextResult\":true,\"action\":3}",
```

Comment:
why are we using commandRunner to add the key? PMC supports the `gpgKeys` see https://github.com/Azure/azure-osconfig/blob/main/src/modules/pmc/README.md

Reply:
Agree, removed this check.